plugins/system/cookiemanager/cookiemanager.php

```diff
@@ -0,0 +1,342 @@
+<?php
+/**
+ * @package     Joomla.Plugin
+ * @subpackage  System.cookiemanager
+ *
+ * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Language\Text;
+use Joomla\CMS\Plugin\CMSPlugin;
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Uri\Uri;
+use Joomla\CMS\WebAsset\WebAssetManager;
+use Joomla\CMS\Component\ComponentHelper;
+
+/**
+ * System plugin to manage cookies.
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class PlgSystemCookiemanager extends CMSPlugin
+{
+	/**
+	 * Application object.
+	 *
+	 * @var    JApplicationCms
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $app;
+
+	/**
+	 * For cookie banner
+	 *
+	 * @var    CookieBanner
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $cookieBanner;
+
+	/**
+	 * For preferences banner
+	 *
+	 * @var    Preferences
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $preferences;
+
+	/**
+	 * Database object
+	 *
+	 * @var    DatabaseDriver
+	 * @since  __DEPLOY_VERSION__
+	 */
+	 protected $db;
+
+	 /**
+	  * For script in DOM
+	  *
+	  * @var    Script
+	  * @since  __DEPLOY_VERSION__
+	  */
+	 protected $script = [];
+
+	 /**
+	  * For scripts in DOM
+	  *
+	  * @var    Scripts
+	  * @since  __DEPLOY_VERSION__
+	  */
+	 protected $scripts;
+
+	 /**
+	  * For cookie category
+	  *
+	  * @var    Category
+	  * @since  __DEPLOY_VERSION__
+	  */
+	 protected $category;
+	/**
+	 * Add assets for the modal.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+
+	 */
+	public function onBeforeCompileHead()
+	{
+		ob_start();
+		ob_implicit_flush(false);
+
+		if (!$this->app->isClient('site'))
+		{
+			return;
+		}
+
+		$this->app->getDocument()->getWebAssetManager()
+			->registerAndUseScript('plg_system_cookiemanager.script', 'plg_system_cookiemanager/cookiemanager.min.js', [], ['defer' => true], ['core'])
+			->registerAndUseStyle('plg_system_cookiemanager.style', 'plg_system_cookiemanager/cookiemanager.min.css');
+
+		$lang = Factory::getLanguage();
+		$lang->load('com_cookiemanager', JPATH_ADMINISTRATOR);
+
+		Text::script('COM_COOKIEMANAGER_PREFERENCES_LESS_BUTTON_TEXT');
+		Text::script('COM_COOKIEMANAGER_PREFERENCES_MORE_BUTTON_TEXT');
+
+		$params = ComponentHelper::getParams('com_cookiemanager');
+		$sitemenu = $this->app->getMenu();
+		$menuitem = $sitemenu->getItem($params->get('policylink'));
+
+		$config = [];
+		$config['expiration'] = $params->get('consent_expiration');
+		$config['position'] = $params->get('modal_position');
+		$this->app->getDocument()->addScriptOptions('config', $config);

```

Comment:
Probably use a better name, which has a prefix from the plugin ("config" is very generic and could be conflicted with something else?)